Scheduled indexing fails in prod

I got this in the logs:

```
19:00:01 INFO  [i.qu.se.ap.in.IndexingService] Scheduled reindex starting...
19:00:01 INFO  [i.qu.se.ap.in.IndexingService] Creating missing indexes
19:00:01 INFO  [i.qu.se.ap.in.IndexingService] Indexing...
19:00:01 INFO  [i.qu.se.ap.in.Rollover] Starting index rollover
19:00:01 INFO  [i.qu.se.ap.in.IndexingService] Clearing database...
19:00:01 WARN  [o.hi.to.sc.in.ExceptionHandlerLoggedImpl] GenerationTarget encountered exception accepting command : Error executing DDL "truncate table Guide" via JDBC [Table "GUIDE" not found (this database is empty);]: org.hibernate.tool.schema.spi.CommandAcceptanceException: Error executing DDL "truncate table Guide" via JDBC [Table "GUIDE" not found (this database is empty);]
	at org.hibernate.tool.schema.internal.exec.GenerationTargetToDatabase.accept(GenerationTargetToDatabase.java:94)
	at org.hibernate.tool.schema.internal.Helper.applySqlString(Helper.java:233)
	at org.hibernate.tool.schema.internal.Helper.applySqlStrings(Helper.java:217)
	at org.hibernate.tool.schema.internal.SchemaTruncatorImpl.truncateFromMetadata(SchemaTruncatorImpl.java:154)
	at org.hibernate.tool.schema.internal.SchemaTruncatorImpl.performTruncate(SchemaTruncatorImpl.java:114)
	at org.hibernate.tool.schema.internal.SchemaTruncatorImpl.doTruncate(SchemaTruncatorImpl.java:91)
	at org.hibernate.tool.schema.internal.SchemaTruncatorImpl.doTruncate(SchemaTruncatorImpl.java:77)
	at org.hibernate.tool.schema.spi.SchemaManagementToolCoordinator.performDatabaseAction(SchemaManagementToolCoordinator.java:301)
	at org.hibernate.tool.schema.spi.SchemaManagementToolCoordinator.lambda$process$5(SchemaManagementToolCoordinator.java:143)
	at java.base/java.util.HashMap.forEach(HashMap.java:1421)
	at org.hibernate.tool.schema.spi.SchemaManagementToolCoordinator.process(SchemaManagementToolCoordinator.java:140)
	at org.hibernate.relational.internal.SchemaManagerImpl.truncateMappedObjects(SchemaManagerImpl.java:83)
	at io.quarkus.search.app.indexing.IndexingService.clearDatabaseWithoutIndexes(IndexingService.java:237)
	at io.quarkus.search.app.indexing.IndexingService.indexAll(IndexingService.java:171)
	at io.quarkus.search.app.indexing.IndexingService.reindex(IndexingService.java:124)
	at io.quarkus.search.app.indexing.IndexingService_Subclass.reindex$$superforward(Unknown Source)
	at io.quarkus.search.app.indexing.IndexingService_Subclass$$function$$2.apply(Unknown Source)
	at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:73)
	at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:62)
	at io.quarkus.arc.impl.ActivateRequestContextInterceptor.invoke(ActivateRequestContextInterceptor.java:124)
	at io.quarkus.arc.impl.ActivateRequestContextInterceptor.aroundInvoke(ActivateRequestContextInterceptor.java:33)
	at io.quarkus.arc.impl.ActivateRequestContextInterceptor_Bean.intercept(Unknown Source)
	at io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:42)
	at io.quarkus.arc.impl.AroundInvokeInvocationContext.perform(AroundInvokeInvocationContext.java:30)
	at io.quarkus.arc.impl.InvocationContexts.performAroundInvoke(InvocationContexts.java:27)
	at io.quarkus.search.app.indexing.IndexingService_Subclass.reindex(Unknown Source)
	at io.quarkus.search.app.indexing.IndexingService.indexOnTime(IndexingService.java:96)
	at io.quarkus.search.app.indexing.IndexingService_ClientProxy.indexOnTime(Unknown Source)
	at io.quarkus.search.app.indexing.IndexingService_ScheduledInvoker_indexOnTime_0d50816a613b249a0196263b6e7d2d4212ae6c23.invokeBean(Unknown Source)
	at io.quarkus.scheduler.common.runtime.DefaultInvoker.invoke(DefaultInvoker.java:24)
	at io.quarkus.scheduler.common.runtime.StatusEmitterInvoker.invoke(StatusEmitterInvoker.java:35)
	at io.quarkus.scheduler.runtime.SimpleScheduler$ScheduledTask.doInvoke(SimpleScheduler.java:430)
	at io.quarkus.scheduler.runtime.SimpleScheduler$ScheduledTask$2.call(SimpleScheduler.java:412)
	at io.quarkus.scheduler.runtime.SimpleScheduler$ScheduledTask$2.call(SimpleScheduler.java:409)
	at io.vertx.core.impl.ContextBase.lambda$executeBlocking$0(ContextBase.java:167)
	at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:277)
	at io.vertx.core.impl.ContextBase.lambda$internalExecuteBlocking$2(ContextBase.java:199)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:587)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: org.h2.jdbc.JdbcSQLSyntaxErrorException: Table "GUIDE" not found (this database is empty); SQL statement:
truncate table Guide [42104-224]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:514)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	at org.h2.message.DbException.get(DbException.java:223)
	at org.h2.message.DbException.get(DbException.java:199)
	at org.h2.command.Parser.getTableOrViewNotFoundDbException(Parser.java:8051)
	at org.h2.command.Parser.getTableOrViewNotFoundDbException(Parser.java:8035)
	at org.h2.command.Parser.readTableOrView(Parser.java:8024)
	at org.h2.command.Parser.readTableOrView(Parser.java:7990)
	at org.h2.command.Parser.parseTruncate(Parser.java:2018)
	at org.h2.command.Parser.parsePrepared(Parser.java:760)
	at org.h2.command.Parser.parse(Parser.java:592)
	at org.h2.command.Parser.parse(Parser.java:564)
	at org.h2.command.Parser.prepareCommand(Parser.java:483)
	at org.h2.engine.SessionLocal.prepareLocal(SessionLocal.java:639)
	at org.h2.engine.SessionLocal.prepareCommand(SessionLocal.java:559)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1166)
	at org.h2.jdbc.JdbcStatement.executeInternal(JdbcStatement.java:245)
	at org.h2.jdbc.JdbcStatement.execute(JdbcStatement.java:231)
	at io.agroal.pool.wrapper.StatementWrapper.execute(StatementWrapper.java:235)
	at org.hibernate.tool.schema.internal.exec.GenerationTargetToDatabase.accept(GenerationTargetToDatabase.java:80)
	... 43 more
```

--- a/src/main/resources/application.properties
+++ b/src/main/resources/application.properties
@@ -33,7 +33,9 @@ quarkus.resteasy-reactive.path=/api
 # Hibernate
 ## We actually don't need persistence, it's just that Hibernate Search requires Hibernate ORM at the moment.
 ## So we just use an in-memory DB.
-quarkus.datasource.jdbc.url=jdbc:h2:mem:searchquarkusio
+## Note DB_CLOSE_DELAY is necessary to avoid the DB stopping and being cleared
+## after some inactivity; see https://github.com/quarkusio/search.quarkus.io/issues/100
+quarkus.datasource.jdbc.url=jdbc:h2:mem:searchquarkusio;DB_CLOSE_DELAY=-1
 quarkus.hibernate-orm.database.generation=drop-and-create
 ## Hibernate Search
 quarkus.hibernate-search-orm.elasticsearch.version=opensearch:2.11

